public/src/components/epicTests/articlesViewedEditor.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import {
+  createStyles, FormControl,
+  FormControlLabel,
+  InputLabel,
+  Switch,
+  Theme, Typography,
+  withStyles,
+  WithStyles
+} from "@material-ui/core";
+import EditableTextField from "../helpers/editableTextField"
+import {onFieldValidationChange, ValidationStatus} from "../helpers/validation";
+import {ArticlesViewedSettings, EpicTest} from "./epicTestsForm";
+
+const isNumber = (value: string): boolean => !Number.isNaN(Number(value));
+
+export const defaultArticlesViewedSettings: ArticlesViewedSettings = {
+  maxViews: null,
+  minViews: 5,
+  periodInWeeks: 8,
+};
+
+const styles = ({ spacing, typography}: Theme) => createStyles({
+  button: {
+    marginTop: '20px'
+  },
+  formControl: {
+    marginTop: spacing(2),
+    marginBottom: spacing(1),
+    display: "block",
+  },
+  selectLabel: {
+    fontSize: typography.pxToRem(22),
+    fontWeight: typography.fontWeightMedium,
+    color: "black"
+  },
+  radio: {
+    paddingTop: "20px",
+    marginBottom: "10px"
+  },
+  articlesViewsContainer: {
+    marginTop: '10px',
+    marginLeft: '20px'
+  }
+});
+
+type ArticleViewsFieldName = 'minViews' | 'maxViews' | 'periodInWeeks';
+
+interface Props extends WithStyles<typeof styles> {
+  articlesViewedSettings?: ArticlesViewedSettings,
+  editMode: boolean,
+  onChange: (articlesViewedSettings?: ArticlesViewedSettings) => void,
+  onValidationChange: (isValid: boolean) => void
+}
+
+interface State {
+  validationStatus: ValidationStatus
+}
+
+class ArticlesViewedEditor extends React.Component<Props, State> {
+  state: State = {
+    validationStatus: {}
+  };
+
+  buildField = (fieldName: ArticleViewsFieldName, label: string, articlesViewedSettings: ArticlesViewedSettings, isRequired: boolean) => {
+    const setting: number | null = articlesViewedSettings[fieldName];
+
+    return (<EditableTextField
+      text={ setting ? setting.toString() : ''}
+      onSubmit={ (value: string) => {
+        if (isNumber(value)) {
+          this.props.onChange(
+            {
+              ...articlesViewedSettings || defaultArticlesViewedSettings,
+              [fieldName]: Number(value)
```

Comment:
@jlieb10 this line is why it's setting the value to 0 when the user deletes the value:
If the field is empty then `value` will be null.
And `Number(null) === 0`

So this should work: `[fieldName]: value !== null ? Number(value) : null`

And to make things clearer, the `if` condition should explicitly allow nulls:
`if (value === null || isNumber(value))`

Reply:
Amazing, thank you